libsForQt5.plasmaMobileGear: 21.08 -> 22.04

dotlambda opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
https://plasma-mobile.org/2021/12/07/plasma-mobile-gear-21-12/
https://plasma-mobile.org/2022/02/09/plasma-mobile-gear-22-02/
https://plasma-mobile.org/2022/04/26/plasma-mobile-gear-22-04/


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @NixOS/qt-kde

dotlambda committed:
libsForQt5.plasmaMobileGear: 21.08 -> 21.12

https://plasma-mobile.org/2021/12/07/plasma-mobile-gear-21-12/

dotlambda committed:
corrosion: unstable-2021-11-23 -> unstable-2022-01-03

dotlambda committed:
libsForQt5.plasmaMobileGear.plasmatube: init at 21.12

samueldr commented:
The reason @mjlbach @peterhoeg are pinged is because neochat is being adopted into the Plasma Mobile Gear release cycle.

samueldr commented:
Checked:

 - [x] Plasma Mobile in Mobile NixOS VM
 - [x] Alligator works
 - [x] Angelfish works
 - [x] (cannot test audio) AudioTube returns results
 - [x] Calculator calculates
 - [x] Calindori starts (seems to work)
 - [x] Clock clocks
 - [x] (cannot test audio) Kasts returns results and downloads podcasts
 - [x] Keysmith seems to work
 - [x] Koko works fine
 - [ ] KTrip has black bars at the bottom, but seems to work otherwise?
 - [x] Phone errors with Modem devices are not found, but otherwise seems to act the part
 - [x] Phonebook seems okay
 - [x] (cannot test) Recorder seems to want to work
 - [x] Spacebar, like Phone, can't be used in the VM, but acts the part
 - [x] Weather works

samueldr commented:
Huh, things were working 25 days ago... but now:

```
CMake Error at CMakeLists.txt:37 (find_package):
  By not providing "FindQCoro.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "QCoro", but
  CMake did not find one.

  Could not find a package configuration file provided by "QCoro" with any of
  the following names:

    QCoroConfig.cmake
    qcoro-config.cmake

  Add the installation prefix of "QCoro" to CMAKE_PREFIX_PATH or set
  "QCoro_DIR" to a directory containing one of the above files.  If "QCoro"
  provides a separate development package or SDK, be sure it has been
  installed.

```

dotlambda committed:
libsForQt5.plasmaMobileGear: 21.12 -> 22.04

https://plasma-mobile.org/2022/02/09/plasma-mobile-gear-22-02/
https://plasma-mobile.org/2022/04/26/plasma-mobile-gear-22-04/

dotlambda committed:
corrosion: unstable-2022-01-03 -> 0.2.1

https://github.com/corrosion-rs/corrosion/blob/v0.2.1/RELEASES.md

dotlambda committed:
libsForQt5.qtmpris: init at 1.0.6

Mindavi reviewed (commented):
Diff LGTM, if I find some time I might try building and running this on my pinephone.

qcoro is broken, that's a shame...

Build failure in audiotube:

```
audiotube> /build/audiotube-22.04/build/src/audiotube_autogen/EWIEGA46WW/../../../../src/multiiterableview.h:9:10: fatal error: span: No such file or directory
audiotube>     9 | #include <span>
audiotube>       |          ^~~~~~
audiotube> compilation terminated.
audiotube> make[2]: *** [src/CMakeFiles/audiotube.dir/build.make:89: src/CMakeFiles/audiotube.dir/audiotube_autogen/mocs_compilation.cpp.o] Error 1
```

Angelfish build is also broken

Will try to build everything tonight, hopefully can report back tomorrow or the day after. (No need to hold up merging)

Aww, OOM killer killed the build, retrying...

dotlambda commented:
Result of `nixpkgs-review pr 153329` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>36 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.alligator</li>
    <li>libsForQt512.angelfish</li>
    <li>libsForQt512.audiotube</li>
    <li>libsForQt512.calindori</li>
    <li>libsForQt512.kalk</li>
    <li>libsForQt512.kasts</li>
    <li>libsForQt512.kclock</li>
    <li>libsForQt512.keysmith</li>
    <li>libsForQt512.koko</li>
    <li>libsForQt512.krecorder</li>
    <li>libsForQt512.ktrip</li>
    <li>libsForQt512.kweather</li>
    <li>libsForQt512.neochat</li>
    <li>libsForQt512.plasma-dialer</li>
    <li>libsForQt512.plasma-phonebook</li>
    <li>libsForQt512.plasma-settings</li>
    <li>libsForQt512.plasmatube</li>
    <li>libsForQt512.spacebar</li>
    <li>libsForQt514.alligator</li>
    <li>libsForQt514.angelfish</li>
    <li>libsForQt514.audiotube</li>
    <li>libsForQt514.calindori</li>
    <li>libsForQt514.kalk</li>
    <li>libsForQt514.kasts</li>
    <li>libsForQt514.kclock</li>
    <li>libsForQt514.keysmith</li>
    <li>libsForQt514.koko</li>
    <li>libsForQt514.krecorder</li>
    <li>libsForQt514.ktrip</li>
    <li>libsForQt514.kweather</li>
    <li>libsForQt514.neochat</li>
    <li>libsForQt514.plasma-dialer</li>
    <li>libsForQt514.plasma-phonebook</li>
    <li>libsForQt514.plasma-settings</li>
    <li>libsForQt514.plasmatube</li>
    <li>libsForQt514.spacebar</li>
  </ul>
</details>
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>corrosion</li>
    <li>libsForQt5.alligator (libsForQt515.alligator ,plasma5Packages.alligator)</li>
    <li>libsForQt5.angelfish (libsForQt515.angelfish ,plasma5Packages.angelfish)</li>
    <li>libsForQt5.audiotube (libsForQt515.audiotube ,plasma5Packages.audiotube)</li>
    <li>libsForQt5.calindori (libsForQt515.calindori ,plasma5Packages.calindori)</li>
    <li>libsForQt5.kalk (libsForQt515.kalk ,plasma5Packages.kalk)</li>
    <li>libsForQt5.kasts (libsForQt515.kasts ,plasma5Packages.kasts)</li>
    <li>libsForQt5.kclock (libsForQt515.kclock ,plasma5Packages.kclock)</li>
    <li>libsForQt5.keysmith (libsForQt515.keysmith ,plasma5Packages.keysmith)</li>
    <li>libsForQt5.koko (libsForQt515.koko ,plasma5Packages.koko)</li>
    <li>libsForQt5.krecorder (libsForQt515.krecorder ,plasma5Packages.krecorder)</li>
    <li>libsForQt5.ktrip (libsForQt515.ktrip ,plasma5Packages.ktrip)</li>
    <li>libsForQt5.kweather (libsForQt515.kweather ,plasma5Packages.kweather)</li>
    <li>libsForQt5.neochat (libsForQt515.neochat ,plasma5Packages.neochat)</li>
    <li>libsForQt5.plasma-dialer (libsForQt515.plasma-dialer ,plasma5Packages.plasma-dialer)</li>
    <li>libsForQt5.plasma-phonebook (libsForQt515.plasma-phonebook ,plasma5Packages.plasma-phonebook)</li>
    <li>libsForQt5.plasma-settings (libsForQt515.plasma-settings ,plasma5Packages.plasma-settings)</li>
    <li>libsForQt5.plasmatube (libsForQt515.plasmatube ,plasma5Packages.plasmatube)</li>
    <li>libsForQt5.qtmpris (libsForQt515.qtmpris ,plasma5Packages.qtmpris)</li>
    <li>libsForQt5.spacebar (libsForQt515.spacebar ,plasma5Packages.spacebar)</li>
    <li>libsForQt512.qtmpris</li>
    <li>libsForQt514.qtmpris</li>
  </ul>
</details>

dotlambda commented:
> Build failure in audiotube

Must be because aarch64 is still on GCC 9.3.0

dotlambda commented:
Result of `nixpkgs-review pr 153329` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>42 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt5.neochat</li>
    <li>libsForQt5.spacebar</li>
    <li>libsForQt512.alligator</li>
    <li>libsForQt512.angelfish</li>
    <li>libsForQt512.audiotube</li>
    <li>libsForQt512.calindori</li>
    <li>libsForQt512.kalk</li>
    <li>libsForQt512.kasts</li>
    <li>libsForQt512.kclock</li>
    <li>libsForQt512.keysmith</li>
    <li>libsForQt512.koko</li>
    <li>libsForQt512.krecorder</li>
    <li>libsForQt512.ktrip</li>
    <li>libsForQt512.kweather</li>
    <li>libsForQt512.neochat</li>
    <li>libsForQt512.plasma-dialer</li>
    <li>libsForQt512.plasma-phonebook</li>
    <li>libsForQt512.plasma-settings</li>
    <li>libsForQt512.plasmatube</li>
    <li>libsForQt512.spacebar</li>
    <li>libsForQt514.alligator</li>
    <li>libsForQt514.angelfish</li>
    <li>libsForQt514.audiotube</li>
    <li>libsForQt514.calindori</li>
    <li>libsForQt514.kalk</li>
    <li>libsForQt514.kasts</li>
    <li>libsForQt514.kclock</li>
    <li>libsForQt514.keysmith</li>
    <li>libsForQt514.koko</li>
    <li>libsForQt514.krecorder</li>
    <li>libsForQt514.ktrip</li>
    <li>libsForQt514.kweather</li>
    <li>libsForQt514.neochat</li>
    <li>libsForQt514.plasma-dialer</li>
    <li>libsForQt514.plasma-phonebook</li>
    <li>libsForQt514.plasma-settings</li>
    <li>libsForQt514.plasmatube</li>
    <li>libsForQt514.spacebar</li>
    <li>libsForQt515.neochat</li>
    <li>libsForQt515.spacebar</li>
    <li>plasma5Packages.neochat</li>
    <li>plasma5Packages.spacebar</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>libsForQt5.angelfish (libsForQt515.angelfish ,plasma5Packages.angelfish)</li>
    <li>libsForQt5.audiotube (libsForQt515.audiotube ,plasma5Packages.audiotube)</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>libsForQt5.alligator (libsForQt515.alligator ,plasma5Packages.alligator)</li>
    <li>libsForQt5.calindori (libsForQt515.calindori ,plasma5Packages.calindori)</li>
    <li>libsForQt5.kalk (libsForQt515.kalk ,plasma5Packages.kalk)</li>
    <li>libsForQt5.kasts (libsForQt515.kasts ,plasma5Packages.kasts)</li>
    <li>libsForQt5.kclock (libsForQt515.kclock ,plasma5Packages.kclock)</li>
    <li>libsForQt5.keysmith (libsForQt515.keysmith ,plasma5Packages.keysmith)</li>
    <li>libsForQt5.koko (libsForQt515.koko ,plasma5Packages.koko)</li>
    <li>libsForQt5.krecorder (libsForQt515.krecorder ,plasma5Packages.krecorder)</li>
    <li>libsForQt5.ktrip (libsForQt515.ktrip ,plasma5Packages.ktrip)</li>
    <li>libsForQt5.kweather (libsForQt515.kweather ,plasma5Packages.kweather)</li>
    <li>libsForQt5.plasma-dialer (libsForQt515.plasma-dialer ,plasma5Packages.plasma-dialer)</li>
    <li>libsForQt5.plasma-phonebook (libsForQt515.plasma-phonebook ,plasma5Packages.plasma-phonebook)</li>
    <li>libsForQt5.plasma-settings (libsForQt515.plasma-settings ,plasma5Packages.plasma-settings)</li>
    <li>libsForQt5.plasmatube (libsForQt515.plasmatube ,plasma5Packages.plasmatube)</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/applications/plasma-mobile/audiotube.nix:
```diff
@@ -16,14 +16,6 @@
 
-  patches = [
-    # Fix compatibility with ytmusicapi 0.19.1
-    (fetchpatch {
```
Please remove the unused fetchpatch from inputs.

SuperSandro2000 commented on pkgs/applications/plasma-mobile/plasmatube.nix:
```diff
@@ -0,0 +1,41 @@
+    description = "Youtube player powered by an invidious server";
+    homepage = "https://invent.kde.org/plasma-mobile/plasmatube";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ ];
```
Thats not great.

Mindavi commented:
Heh, does just adding gcc11 work? I wouldn't expect so...

Mindavi commented:
@ofborg build libsForQt515.audiotube

SuperSandro2000 commented:
> Heh, does just adding gcc11 work? I wouldn't expect so...

I think we want to use gcc11Stdenv instead.

dotlambda commented:
> Heh, does just adding gcc11 work? I wouldn't expect so...

It does:

```
-- The C compiler identification is GNU 11.3.0
```



> I think we want to use gcc11Stdenv instead.

I don't know how because of the special `libsForQt5.mkDerivation`. Just adding `gcc11` to `nativeBuildInputs` was a simple solution and it works.

Mindavi commented:
So the diff LGTM, not sure if can properly test otherwise since I'm having trouble with getting any useful output when trying to enable a display manager. It's probably due to an unrelated bug somewhere :(.

Just to have it noted somewhere, for some reason the users belonging to the modules (sddm, lightdm) are not created at all...

dotlambda commented:
Result of `nixpkgs-review pr 153329` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>32 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.alligator</li>
    <li>libsForQt512.angelfish</li>
    <li>libsForQt512.audiotube</li>
    <li>libsForQt512.calindori</li>
    <li>libsForQt512.kalk</li>
    <li>libsForQt512.kasts</li>
    <li>libsForQt512.kclock</li>
    <li>libsForQt512.keysmith</li>
    <li>libsForQt512.koko</li>
    <li>libsForQt512.krecorder</li>
    <li>libsForQt512.ktrip</li>
    <li>libsForQt512.kweather</li>
    <li>libsForQt512.plasma-dialer</li>
    <li>libsForQt512.plasma-phonebook</li>
    <li>libsForQt512.plasma-settings</li>
    <li>libsForQt512.plasmatube</li>
    <li>libsForQt514.alligator</li>
    <li>libsForQt514.angelfish</li>
    <li>libsForQt514.audiotube</li>
    <li>libsForQt514.calindori</li>
    <li>libsForQt514.kalk</li>
    <li>libsForQt514.kasts</li>
    <li>libsForQt514.kclock</li>
    <li>libsForQt514.keysmith</li>
    <li>libsForQt514.koko</li>
    <li>libsForQt514.krecorder</li>
    <li>libsForQt514.ktrip</li>
    <li>libsForQt514.kweather</li>
    <li>libsForQt514.plasma-dialer</li>
    <li>libsForQt514.plasma-phonebook</li>
    <li>libsForQt514.plasma-settings</li>
    <li>libsForQt514.plasmatube</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>corrosion</li>
    <li>libsForQt5.alligator</li>
    <li>libsForQt5.angelfish</li>
    <li>libsForQt5.audiotube</li>
    <li>libsForQt5.calindori</li>
    <li>libsForQt5.kalk</li>
    <li>libsForQt5.kasts</li>
    <li>libsForQt5.kclock</li>
    <li>libsForQt5.keysmith</li>
    <li>libsForQt5.koko</li>
    <li>libsForQt5.krecorder</li>
    <li>libsForQt5.ktrip</li>
    <li>libsForQt5.kweather</li>
    <li>libsForQt5.plasma-dialer</li>
    <li>libsForQt5.plasma-phonebook</li>
    <li>libsForQt5.plasma-settings</li>
    <li>libsForQt5.plasmatube</li>
    <li>libsForQt5.qtmpris</li>
    <li>libsForQt512.qtmpris</li>
    <li>libsForQt514.qtmpris</li>
  </ul>
</details>

dotlambda commented:
@samueldr Any objection to me merging this?

dotlambda commented:
@andrevmatos Do you want me to add you as maintainer of something?

andrevmatos commented:
@dotlambda I just tried my hand on this update to split the fixes broken by my try at #173266 ; eventually, I'd like to join @NixOS/qt-kde , as like to help keep KDE packages up to date; I don't have a direct interest on any specific package on this PR, but feel free to pull me in however I can help =)
